sdcm/utils/data_validator.py

```diff
@@ -469,6 +498,168 @@ def list_of_view_names_for_update_test(self):
+            Optional[DataForValidation]:
+        # views_set[0] - view name with rows before update
+        # views_set[1] - view name with rows after update
+        # views_set[2] - view name with all expected partition keys
+        # views_set[3] - do perform validation for the view or not
+        partition_keys = ', '.join(self.base_table_partition_keys)
+
+        before_update_rows = self.longevity_self_object.fetch_all_rows(
+            session=session, default_fetch_size=self.DEFAULT_FETCH_SIZE,
+            statement=f"SELECT {partition_keys} FROM {views_set[0]}",
+            verbose=not during_nemesis)
+        if not before_update_rows:
+            DataValidatorEvent.UpdatedRowsValidator(
+                severity=Severity.WARNING,
+                message=f"Can't validate updated rows. Fetch all rows from {views_set[0]} failed. "
+                        f"See error above in the sct.log"
+            ).publish()
+            return None
+
+        after_update_rows = self.longevity_self_object.fetch_all_rows(
+            session=session, default_fetch_size=self.DEFAULT_FETCH_SIZE,
+            statement=f"SELECT {partition_keys} FROM {views_set[1]}",
+            verbose=not during_nemesis)
+        if not after_update_rows:
+            DataValidatorEvent.UpdatedRowsValidator(
+                severity=Severity.WARNING,
+                message=f"Can't validate updated rows. Fetch all rows from {views_set[1]} failed. "
+                        f"See error above in the sct.log"
+            ).publish()
+            return None
+
+        expected_rows = self.longevity_self_object.fetch_all_rows(
+            session=session, default_fetch_size=self.DEFAULT_FETCH_SIZE,
+            statement=f"SELECT {partition_keys} FROM {views_set[2]}",
+            verbose=not during_nemesis)
+        if not expected_rows:
+            DataValidatorEvent.UpdatedRowsValidator(
+                severity=Severity.WARNING,
+                message=f"Can't validate updated row. Fetch all rows from {views_set[2]} failed. "
+                        f"See error above in the sct.log"
+            ).publish()
+            return None
+
+        return DataForValidation(views=views_set,
+                                 actual_data=sorted(before_update_rows + after_update_rows),
+                                 expected_data=sorted(expected_rows),
+                                 before_update_rows=before_update_rows,
+                                 after_update_rows=after_update_rows)
+
+    @staticmethod
+    def save_data_for_debugging(data_for_validation: DataForValidation) -> str:
+        log_dir = TestConfig().logdir()
+        logdir = os.path.join(log_dir, "lwt_validator_data_for_debug")
+        os.makedirs(logdir, exist_ok=True)
+
+        with open(os.path.join(logdir, f"{data_for_validation.views[0]}_debug.json"), "w", encoding='utf8') \
```

Comment:
we assume this would happen only once in a test ?

cause if not, we need to put something more into the names, like timestamp, or uuid.

Reply:
added uuid to the file name